services/subscriptions/subscriptions.go

```diff
@@ -0,0 +1,80 @@
+package subscriptions
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type Subscriptions struct {
+	mu          sync.Mutex
+	subs        map[SubscriptionID]*Subscription
+	checkPeriod time.Duration
+}
+
+func NewSubscriptions(period time.Duration) *Subscriptions {
+	return &Subscriptions{
+		mu:          sync.Mutex{},
```

Comment:
zero value is unlocked mutex. usually there is no reason to initialize it explicitly. the only situation i can come up with is when you pass object by value and want to avoid copying mutex